DDDEastAnglia/Controllers/AccountController.cs

```diff
@@ -16,18 +17,41 @@ namespace DDDEastAnglia.Controllers
     public class AccountController : Controller
     {
         private readonly IUserProfileRepository userProfileRepository;
-
-        public AccountController(IUserProfileRepository userProfileRepository)
+        private readonly IBuild<LoginMethod, LoginMethodViewModel> loginMethodViewModelBuilder;
+        private readonly ExternalLoginsProvider externalLoginsProvider;
+        private readonly IOAuthClientInfo oAuthClientInfo;
+
+        public AccountController(IUserProfileRepository userProfileRepository, 
```

Comment:
Ok. Is there a service we can pull out to abstract over some of these dependencies for the external login actions? I can see the controller needing this façade service and the `IBuild<LoginMethod, LoginMethodViewModel>`, but not much else. 

Ugh, actually, this looks like a fat controller that needs breaking up, but this isn't the place for it.